src/outputs/parthenon_opmd.cpp

```diff
@@ -0,0 +1,517 @@
+  // but an interation) This just describes the pattern of the filename. The correct file
+  // will be accessed through the iteration idx below. The file suffix maps to the chosen
+  // backend.
+  // TODO(pgrete) add final and now logic
+  Series series =
+      Series(output_params.file_basename + "." + output_params.file_id + ".%05T.bp",
+             Access::CREATE);
+
+  // TODO(pgrete) How to handle downstream info, e.g.,  on how/what defines a vector?
+  // TODO(pgrete) Should we update for restart or only set this once? Or make it per
+  // iteration?
+  // ... = pin->GetString(output_params.block_name, "actions_file");
+  series.setAuthor("My Name <mail@addre.es");
+  series.setComment("Hello world!");
+  series.setMachine("bla");
+  series.setSoftware("Parthenon + Downstream info");
+  series.setDate("2024-02-29");
+
+  // TODO(pgrete) Units?
+
+  // TODO(pgrete) We probably want this for params!
+  series.setAttribute("bla", true);
+
+  // In line with existing outputs, we write one file per iteration/snapshot
+  series.setIterationEncoding(openPMD::IterationEncoding::fileBased);
+
+  // open iteration (corresponding to a timestep in OpenPMD naming)
+  // TODO(pgrete) fix iteration name <-> file naming
+  auto it = series.iterations[output_params.file_number];
+  it.open(); // explicit open() is important when run in parallel
+
+  auto const &first_block = *(pm->block_list.front());
+
+  // TODO(?) in principle, we could abstract this to a more general WriteAttributes place
+  // and reuse for hdf5 and OpenPMD output with corresponing calls
+  // -------------------------------------------------------------------------------- //
+  //   WRITING ATTRIBUTES                                                             //
+  // -------------------------------------------------------------------------------- //
+
+  // Note, that profiling is likely skewed as data is actually written to disk/flushed
+  // only later.
+  Kokkos::Profiling::pushRegion("write Attributes");
+  // First the ones required by the OpenPMD standard
+  if (tm != nullptr) {
+    it.setTime(tm->time);
+    it.setDt(tm->dt);
+    it.setAttribute("NCycle", tm->ncycle);
+  } else {
+    it.setTime(-1.0);
+    it.setDt(-1.0);
+  }
+  { // FIXME move this to dump params
+    PARTHENON_INSTRUMENT_REGION("Dump Params");
+    const auto view_d =
+        Kokkos::View<Real **, Kokkos::DefaultExecutionSpace>("blub", 5, 3);
+    // Map a view onto a host allocation (so that we can call deep_copy)
+    auto host_vec = std::vector<Real>(view_d.size());
+    Kokkos::View<Real **, Kokkos::HostSpace, Kokkos::MemoryTraits<Kokkos::Unmanaged>>
+        view_h(host_vec.data(), view_d.extent_int(0), view_d.extent_int(1));
+    Kokkos::deep_copy(view_h, view_d);
+    it.setAttribute("blub", host_vec);
+
+    for (const auto &[key, pkg] : pm->packages.AllPackages()) {
+      // WriteAllParams<bool>(pkg, &it); // check why this (vector of bool) doesn't work
```

Comment:
Is it just `std::vector<bool>` that doesn't work? That's probably because `std::vector<bool>` has a particular template specialization in the stl that is weirdly incompatible with generic code. Probably need to write a pointer-based array of bool instead.